editor/editor_themes.cpp

```diff
@@ -829,7 +829,7 @@ Ref<Theme> create_editor_theme(const Ref<Theme> p_theme) {
 	theme->set_constant("side_margin", "TabContainer", 0);
 	theme->set_icon("tab", "TextEdit", theme->get_icon("GuiTab", "EditorIcons"));
 	theme->set_color("font_color", "TextEdit", font_color);
-	theme->set_color("caret_color", "TextEdit", highlight_color);
+	theme->set_color("caret_color", "TextEdit", mono_color); 
```

Comment:
It'd make more sense to use `font_color` for consistency, as it's the color used by the cursor in LineEdit:

```suggestion
	theme->set_color("caret_color", "TextEdit", font_color);
```